modules/imgproc/src/resize.cpp

```diff
@@ -1481,10 +1481,112 @@ typedef VResizeNoVec VResizeLanczos4Vec_32f;
 
 #endif
 
+#if CV_VSX
+
+/*
+    Deinterleave two vectors into the same sized type.
+
+    A1 B1 A2 B2 .... A4 B4
+    and return expanded int32x4 of the form:
+    A1 ... A4
+    B1 ... B4
+*/
+template<typename T, typename VT>
+void inline v_load_expand_deinterlace(const T *ptr, VT &out_even, VT &out_odd)
+{
+    v_load_deinterleave(ptr, out_even, out_odd);
+}
+
+/*
+    Similar to above, but we must expand to the larger type and deinterleave the
+    expanded values.
+
+    The quick expand and deinterlace is PPC specific. This operation may not
+    port efficiently to other vector engines.
+*/
+template<>
+void inline v_load_expand_deinterlace(const short *ptr, v_int32x4 &out_even, v_int32x4 &out_odd)
+{
+    static const vec_uchar16 perm = {0,1,4,5,8,9,12,13,2,3,6,7,10,11,14,15};
+    v_int16x8 in = v_load(ptr);
+    in.val = vec_perm( in.val, in.val, perm );
+    out_even = v_int32x4(vec_unpackh(in.val));
+    out_odd = v_int32x4(vec_unpackl(in.val));
+}
+
+template<typename ST, typename DT, typename AT, typename DVT>
+struct HResizeLinearVec_X4
+{
+    int operator()(const uchar** _src, uchar** _dst, int count, const int* xofs,
+        const uchar* _alpha, int, int, int cn, int, int xmax) const
+    {
+        const ST **src = (const ST**)_src;
+        const AT *alpha = (const AT*)_alpha;
+        DT **dst = (DT**)_dst;
+        const int nlanes=4;
+        int dx = 0, k = 0;
+
+        for( ; k <= (count - 2); k+=2 )
+        {
+            const ST *S0 = src[k];
+            DT *D0 = dst[k];
+            const ST *S1 = src[k+1];
+            DT *D1 = dst[k+1];
+
+            for( dx = 0; dx < (xmax - nlanes); dx+=nlanes )
```

Comment:
It would be better to use `dx <= (xmax - nlanes)` here to process one more vector.
The same at L#1558